drivers/sensor/ags10/ags10.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Balthazar Deliers
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT aosong_ags10
+
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/sys/byteorder.h>
+
+#include "ags10.h"
+
+LOG_MODULE_REGISTER(AGS10, CONFIG_SENSOR_LOG_LEVEL);
+
+static int ags10_write(const struct device *dev, uint8_t cmd, uint8_t* data, uint8_t bytes)
+{
+	const struct ags10_config *conf = dev->config;
+	uint8_t send_buf[6] = {0};
+
+	send_buf[0] = cmd;
+	memcpy(&send_buf[1], data, bytes);
+
+	return i2c_write_dt(&conf->bus, send_buf, bytes);
+}
+
+static int ags10_read(const struct device *dev, uint8_t cmd, uint8_t *data, uint8_t rx_bytes)
+{
+	const struct ags10_config *conf = dev->config;
+
+	uint8_t recv_buf[5] = {0};
+	int ret = i2c_write_read_dt(&conf->bus,
+				    &cmd,
+				    sizeof(cmd),
+				    &recv_buf,
+				    rx_bytes);
+	if (ret < 0) {
+		return ret;
+	}
+
+	memcpy(data, recv_buf, rx_bytes);
+
+	return 0;
+}
+
+/*
+ * @brief Calculation provided by the manufacturer in the datasheet part 3.7
+ * @param data  Data buffer on which the CRC needs to be calculated
+ * @param size  Data size in bytes
+ */
+static uint8_t ags10_crc(uint8_t *data, uint8_t size)

```

Comment:
Could use [Zephyr's CRC API](https://docs.zephyrproject.org/apidoc/latest/group__crc.html#gaa614e73ee7484ca0424fa7d14a54bbb6) here. I think `crc8()` with `0x31` as the polynomial and `0xFF` as the initial value would be equivalent.

Reply:
done